src/algorithm/map_coords.rs

```diff
@@ -59,16 +109,37 @@ impl<T: CoordinateType, NT: CoordinateType> MapCoords<T, NT> for Point<T> {
     }
 }
 
-impl<T: CoordinateType> MapCoordsInplace<T> for Point<T> {
+impl<T: CoordinateType, NT: CoordinateType> MapCoordsFallible<T, NT> for Point<T> {
```

Comment:
instead of having all these explicit `MapCoordsFallible` `impl`s, what if we had a blanket `impl` like this (i haven't tried compiling it so it might be wrong):

```rust
impl<T: CoordinateType, NT: CoordinateType, G: MapCoords<T, NT>> MapCoordsFallible<T, NT> for G {
    type Output = G::Output;

    fn map_coords_fallible(
        &self,
        func: &Fn(&(T, T)) -> Result<(NT, NT), Error>,
    ) -> Result<Self::Output, Error> {
        Ok(self.map_coords(func))
    }
}
```

Reply:
I don't *think* we can implement it in terms of `map_coords`, because of the different input function signatures:
`map_coords` is `func: &Fn(&(T, T)) -> (NT, NT)`  
`map_coords_fallible` is `func: &Fn(&(T, T)) -> Result<(NT, NT), Error>`

And (I think) because we're also missing a `try!` or `?` to check for an early return, somewhere.

Reply:
yep, you're right 👍 

i don't think there's a way to make this work since we can't return from the outer function within the closure if `func` returs an error